test/e2e/federated-namespace.go

```diff
@@ -88,40 +105,54 @@ var _ = framework.KubeDescribe("Federation namespace [Feature:Federation]", func
 			})
 			framework.ExpectNoError(err, "Not all namespaces created")
 
-			err = f.FederationClientset_1_4.Core().Namespaces().Delete(ns.Name, &api.DeleteOptions{})
-			framework.ExpectNoError(err, "Failed to delete namespace %s", ns.Name)
+			// TODO: set wait to true once NS controller is fixed.
+			deleteAllTestNamespaces(
+				func() (*api_v1.NamespaceList, error) {
+					return f.FederationClientset_1_4.Core().Namespaces().List(api.ListOptions{})
+				},
+				func(ns *api_v1.Namespace) error {
+					_, err := f.FederationClientset_1_4.Core().Namespaces().Update(ns)
+					return err
+				},
+				false)
 		})
 	})
 })
 
-func deleteAllTestNamespacesFed14(clientset *federation_release_1_4.Clientset) {
-	list, err := clientset.Core().Namespaces().List(api.ListOptions{})
+func deleteAllTestNamespaces(lister func() (*api_v1.NamespaceList, error), updater func(*api_v1.Namespace) error, waitForDeletion bool) {
+	list, err := lister()
 	if err != nil {
 		framework.Failf("Failed to get all namespaes: %v", err)
 		return
 	}
 	for _, namespace := range list.Items {
-		if strings.HasPrefix(namespace.Name, namespacePrefix) {
-			err := clientset.Core().Namespaces().Delete(namespace.Name, &api.DeleteOptions{})
+		if strings.HasPrefix(namespace.Name, namespacePrefix) && namespace.DeletionTimestamp != nil {
+			namespace.DeletionTimestamp = &unversioned.Time{Time: time.Now()}
```

Comment:
Actually, you shouldn't manually set DeletionTimestamp (at least in the base cluster).

In the base clusters, calling Delete() actually sets the DeletionTimestamp (calling Delete will not delete the namespace - it will just set DeletionTimestamp).

I think that we generally go towards the same logic in federation.